### PR TITLE
Fix/appid validation v3

### PR DESCRIPTION
**Pull request by akila95fernando**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java [hunk 1]
```diff
@@ -260,9 +261,13 @@ public KeyManagerConfiguration getKeyManagerConfiguration() throws APIManagement
     @Override
     public OAuthApplicationInfo mapOAuthApplication(OAuthAppRequest oAuthAppRequest) throws APIManagementException {
         String consumerKey = oAuthAppRequest.getOAuthApplicationInfo().getClientId();
-
+        OAuthApplicationInfo clientInfo = null;
         if (StringUtils.isNotBlank(consumerKey)) {
-            OAuthApplicationInfo clientInfo = retrieveApplication(consumerKey);
+            try{
```
Comment (tharikaGitHub) [inline components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java:266]: Let's format the code in the try catch block.

File: components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java [hunk 2]
```diff
@@ -260,9 +261,13 @@ public KeyManagerConfiguration getKeyManagerConfiguration() throws APIManagement
...
+            clientInfo = this.getOAuthApplicationInfo(appClient.getApplicationByAppId(consumerKey));
```
Comment (tharikaGitHub) [inline components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java:267]: If the result returned from `appClient.getApplicationByAppId(consumerKey)` is null, there will be a null pointer exception thrown from `this.getOAuthApplicationInfo` method. Let's fix it to first check for null and invoke `getOAuthApplicationInfo()`.

File: components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java [hunk 3]
```diff
@@ -260,9 +261,18 @@ public KeyManagerConfiguration getKeyManagerConfiguration() throws APIManagement
     @Override
     public OAuthApplicationInfo mapOAuthApplication(OAuthAppRequest oAuthAppRequest) throws APIManagementException {
         String consumerKey = oAuthAppRequest.getOAuthApplicationInfo().getClientId();
-
+        OAuthApplicationInfo clientInfo = null;
         if (StringUtils.isNotBlank(consumerKey)) {
-            OAuthApplicationInfo clientInfo = retrieveApplication(consumerKey);
+            try {
+                ClientInformation clientInformation = appClient.getApplicationByAppId(consumerKey);
+                if ( clientInformation != null ) {
```
Comment (chamilaadhi) [inline components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java:268]: Can you format this code section please?

Comment (akila95fernando) [reply components/azure.key.manager/src/main/java/org/wso2/azure/client/AzureADClient.java]: done via https://github.com/wso2-extensions/apim-km-azure-ad/pull/8/commits/2ccef20a76c3a59530c787eb9c43dcd5b5d2e835

**Review by chamilaadhi — APPROVED** [review]

**Pull request merged**